### PR TITLE
fix(grpc-transcode): pb_option should be set on the route

### LINKED ISSUE #7025: bug: grpc-transcode's pb_option is set on the proto level
### Current Behavior

The `pb.option` method in lua-protobuf is set on the pb_State object, which is mapped from the proto object in APISIX. Therefore, setting the option will affect other routes which share the same proto.

Setting the option globally won't affect grpc-transcode as each proto has its own pb_State.

### Expected Behavior

_No response_

### Error Logs

_No response_

### Steps to Reproduce

The test case below shows it:
```
=== TEST 12: set routes(id: 2)
--- ONLY
--- config
    location /t {
        content_by_lua_block {
            local t = require("lib.test_admin").test
            local code, body = t('/apisix/admin/routes/1',
                ngx.HTTP_PUT,
                [[{
                    "methods": ["GET"],
                    "uri": "/grpc_plus2",
                    "plugins": {
                        "grpc-transcode": {
                            "proto_id": "1",
                            "service": "helloworld.Greeter",
                            "method": "Plus"
                        }
                    },
                    "upstream": {
                        "scheme": "grpc",
                        "type": "roundrobin",
                        "nodes": {
                            "127.0.0.1:50051": 1
                        }
                    }
                }]]
            )
            local code, body = t('/apisix/admin/routes/2',
                ngx.HTTP_PUT,
                [[{
                    "methods": ["GET"],
                    "uri": "/grpc_plus",
                    "plugins": {
                        "grpc-transcode": {
                            "proto_id": "1",
                            "service": "helloworld.Greeter",
                            "method": "Plus",
                            "pb_option":["int64_as_string", "enum_as_name"]
                        }
                    },
                    "upstream": {
                        "scheme": "grpc",
                        "type": "roundrobin",
                        "nodes": {
                            "127.0.0.1:50051": 1
                        }
                    }
                }]]
            )

            if code >= 300 then
                ngx.status = code
            end
            ngx.say(body)

            local http = require "resty.http"
            local uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/grpc_plus?a=1&b=2251799813685260"
            local httpc = http.new()
            local res, err = httpc:request_uri(uri)
            ngx.say(res.body)
            local uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/grpc_plus2?a=1&b=2251799813685260"
            local httpc = http.new()
            local res, err = httpc:request_uri(uri)
            ngx.say(res.body)
        }
    }
--- request
GET /t
--- response_body
passed
--- no_error_log
[error]
```

The route configurations of `/grpc_plus` & `/grpc_plus2` are different, but now they provide the same result because of the pollution.

### Environment

- APISIX version (run `apisix version`): 2.13.1
- Operating system (run `uname -a`): Linux DESKTOP-2022Q8F-wsl 4.19.128-microsoft-standard
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): nginx version: openresty/1.19.9.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Description

Fixes #7025

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/grpc-transcode/util.lua [hunk 1]
```diff
@@ -51,6 +53,35 @@ function _M.find_method(proto, service, method)
...
+        end
+
+        local same = true
+        table.sort(options)
+        for i, v in ipairs(options) do
+            if cur_opts[i] ~= v then
+                same = false
+                break
+            end
+        end
```
Comment (bisakhmondal) [inline apisix/plugins/grpc-transcode/util.lua:70]: I assume the extra check of pointer and values are for perf improvements. But just a small heads up, this check will fail on the second time the `set_option` method is called and on the first call, the `cur_opts` was nil.

In that scenario, at the first run, the unsorted table will be cached - L81.
L66 will try to match with the sorted table `options`. 

Anyway, just a small nitpicking indeed : )

**Review by bisakhmondal — COMMENTED** [review]
In general look good to me. Thanks!

**Review by tokers — DISMISSED** [review]

**Review by tzssangglass — DISMISSED** [review]

File: apisix/plugins/grpc-transcode/util.lua (hunk 1 above)
Comment (spacewander) [reply apisix/plugins/grpc-transcode/util.lua:71]: Good catch!

**Review by bisakhmondal — APPROVED** [review]
LGTM. Thanks!

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**